cpp/src/io/csv/csv_reader.cu

```diff
@@ -190,6 +195,80 @@ std::string stringType(gdf_dtype dt){
 
 }
 
+/**
+ * reads contents of an arrow::io::RandomAccessFile in a char * buffer up to the number of bytes specified in bytes_to_read
+ * for non local filesystems where latency and availability can be an issue it will retry until it has exhausted its the read attemps and empty reads that are allowed
+ */
+gdf_error read_file_into_buffer(std::shared_ptr<arrow::io::RandomAccessFile> file, int64_t bytes_to_read, uint8_t* buffer, int total_read_attempts_allowed, int empty_reads_allowed){
```

Comment:
So would this also work with an Arrow HDFS file?

Reply:
Yeap! That's correct. As soon as we test with hdfs filesystem we will set this PR for REVIEW.

Reply:
Awesome.  That's pretty exciting.